clients/client-shell/cmds/validate-json/validate.go

```diff
@@ -0,0 +1,71 @@
+// Package validatejson impliments validate-json command
+package validatejson
+
+import (
+	"errors"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/taskcluster/taskcluster/v44/clients/client-shell/cmds/root"
+	js "github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	Command = &cobra.Command{
+		Use:   "validate-json <json-schema> <json-file>",
+		Short: "Validate json file by provided json-schema",
+		RunE:  validate,
+	}
+
+	log = root.Logger
+)
+
+func init() {
+	root.Command.AddCommand(Command)
+}
+
+// Takes json-schema & json as input which should be in the following 2 formats
+// 1. https://community-tc.services.mozilla.com/references/schemas/<some-schema>.json
+// 2. file:///home/user/<some-schema>.json
+func validate(cmd *cobra.Command, args []string) error {
+	schema := schemaLoader(args[0])
+	document := schemaLoader(args[1])
+
+	result, err := js.Validate(schema, document)
+
+	if err != nil {
+		log.Panic(err.Error())
```

Comment:
I tried validating with a YAML from the repository, since the original bug seems to mention this schema:

```
./taskcluster validate-json https://raw.githubusercontent.com/taskcluster/taskcluster/main/workers/docker-worker/schemas/v1/payload.yml test.json
```

I got:

```
PANI invalid character '$' looking for beginning of value 
panic: (*logrus.Entry) (0x1581600,0xc0000de0e0)

goroutine 1 [running]:
github.com/sirupsen/logrus.(*Entry).log(0xc0000de070, 0x0, 0xc000264100, 0x34)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/sirupsen/logrus@v1.8.1/entry.go:259 +0x2e2
github.com/sirupsen/logrus.(*Entry).Log(0xc0000de070, 0xc000000000, 0xc0004bfd60, 0x1, 0x1)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/sirupsen/logrus@v1.8.1/entry.go:293 +0x86
github.com/sirupsen/logrus.(*Logger).Log(0x19c60e0, 0x0, 0xc0004bfd60, 0x1, 0x1)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/sirupsen/logrus@v1.8.1/logger.go:198 +0x7d
github.com/sirupsen/logrus.(*Logger).Panic(...)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/sirupsen/logrus@v1.8.1/logger.go:247
github.com/taskcluster/taskcluster/v44/clients/client-shell/cmds/validate-json.validate(0x19c98c0, 0xc0000bb4e0, 0x2, 0x2, 0x0, 0x0)
	/Users/john/src/taskcluster/clients/client-shell/cmds/validate-json/validate.go:38 +0x351
github.com/spf13/cobra.(*Command).execute(0x19c98c0, 0xc0000bb480, 0x2, 0x2, 0x19c98c0, 0xc0000bb480)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000c6280, 0x0, 0x80400, 0xc000098058)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/john/.gvm/pkgsets/go1.13.7/global/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
	/Users/john/src/taskcluster/clients/client-shell/main.go:15 +0x32
```

This seems like a rude amount of traceback for a parsing error, but it also lead me directly to the problem.

Reply:
It can be taken care of in a sane way by just returning the error & not throwing a panic (reason for the whole traceback). I've added the appropriate changes.

Reply:
thank you so much vnzongzna